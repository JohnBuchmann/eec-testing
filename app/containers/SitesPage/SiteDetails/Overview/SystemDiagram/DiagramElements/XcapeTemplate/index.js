/**
 * XCape template
 */

import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { makeStyles } from '@material-ui/core/styles';
import { Colors } from 'Theme';
import { getSystemDiagramLabelingConfiguration } from 'Utils/systemDiagram';
import messages from './messages';
import Labeling from '../Labeling';

/**
 * @property {Object} useStyles Stores the styles
 */
const useStyles = makeStyles({
  autoSyncDCBusTitle: {
    fontFamily: 'Inter, sans-serif',
    fontSize: '12px',
    fontWeight: '300',
    lineHeight: 'normal',
    letterSpacing: '.6px',
    textTransform: 'uppercase',
    color: Colors.white,
  },
});

/**
 * Displays XCape layer template
 * @param {*} props
 */
function XcapeTemplate(props) {
  const { formatMessage } = props.intl;
  const { autoSyncDCBusTitle } = messages;

  const classes = useStyles();

  return (
    <>
      {/* Micro Grid Consumption Section frame */}
      <rect
        width="506"
        height="436"
        x="147"
        y="60"
        rx="40"
        ry="40"
        fill="none"
        stroke={Colors.white}
        strokeWidth="1"
      />
      {/* Micro Grid Production Section frame */}
      <rect
        width="489"
        height="436"
        x="660"
        y="61"
        rx="40"
        ry="40"
        fill="none"
        stroke={Colors.white}
        strokeWidth="1"
      />
      <path
        stroke={Colors.white}
        fill={Colors.nandor}
        d="M464,267,h572,q5,0 5, 5,v12,q0,5 -5,5h-572z"
      />
      <text x={863} y={282} className={classes.autoSyncDCBusTitle}>
        {formatMessage(autoSyncDCBusTitle)}
      </text>

      {getSystemDiagramLabelingConfiguration().map((label) => (
        <Labeling key={label.name} {...label} />
      ))}
    </>
  );
}

XcapeTemplate.propTypes = {
  intl: PropTypes.any.isRequired,
};

export default injectIntl(XcapeTemplate);
