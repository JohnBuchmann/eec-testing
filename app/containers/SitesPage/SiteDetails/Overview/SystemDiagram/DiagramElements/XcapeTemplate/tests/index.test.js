import { mount } from 'enzyme';
import React from 'react';
import { IntlProvider } from 'react-intl';
import XcapeTemplate from '..';

describe('XcapeTemplate', () => {
  it('should display Xcape Template', () => {
    const actualComponent = mount(
      <svg xmlns="http://www.w3.org/2000/svg">
        <IntlProvider locale="en">
          <XcapeTemplate />
        </IntlProvider>
      </svg>
    );

    expect(actualComponent).toMatchSnapshot();
  });
});
