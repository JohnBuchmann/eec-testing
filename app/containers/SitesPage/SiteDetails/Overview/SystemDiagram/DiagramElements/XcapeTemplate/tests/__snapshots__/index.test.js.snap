// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XcapeTemplate should display Xcape Template 1`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
>
  <IntlProvider
    locale="en"
  >
    <InjectIntl(XcapeTemplate)>
      <XcapeTemplate
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {},
            "now": [Function],
            "onError": [Function],
            "textComponent": "span",
            "timeZone": null,
          }
        }
      >
        <rect
          fill="none"
          height="436"
          rx="40"
          ry="40"
          stroke="#ffffff"
          strokeWidth="1"
          width="506"
          x="147"
          y="60"
        />
        <rect
          fill="none"
          height="436"
          rx="40"
          ry="40"
          stroke="#ffffff"
          strokeWidth="1"
          width="489"
          x="660"
          y="61"
        />
        <path
          d="M464,267,h572,q5,0 5, 5,v12,q0,5 -5,5h-572z"
          fill="#515e51"
          stroke="#ffffff"
        />
        <text
          className="makeStyles-autoSyncDCBusTitle-1"
          x={863}
          y={282}
        >
          dc-bus
        </text>
        <Labeling
          description="To/From"
          label="T/F"
          x={360}
          y={530}
        >
          <desc>
            Label
          </desc>
          <text
            className="makeStyles-label-3"
            x={360}
            y={530}
          >
            <tspan
              className="makeStyles-labelTitle-4"
            >
              T/F
              :
            </tspan>
            <tspan
              dx={5}
            >
              To/From
            </tspan>
          </text>
        </Labeling>
        <Labeling
          description="Actual"
          label="ACT"
          x={750}
          y={530}
        >
          <desc>
            Label
          </desc>
          <text
            className="makeStyles-label-3"
            x={750}
            y={530}
          >
            <tspan
              className="makeStyles-labelTitle-4"
            >
              ACT
              :
            </tspan>
            <tspan
              dx={5}
            >
              Actual
            </tspan>
          </text>
        </Labeling>
        <Labeling
          description="Device Capacity"
          label="CAP"
          x={951}
          y={530}
        >
          <desc>
            Label
          </desc>
          <text
            className="makeStyles-label-3"
            x={951}
            y={530}
          >
            <tspan
              className="makeStyles-labelTitle-4"
            >
              CAP
              :
            </tspan>
            <tspan
              dx={5}
            >
              Device Capacity
            </tspan>
          </text>
        </Labeling>
      </XcapeTemplate>
    </InjectIntl(XcapeTemplate)>
  </IntlProvider>
</svg>
`;
